pinot-batch-ingestion/pinot-standalone/src/main/java/org/apache/pinot/ingestion/standalone/SegmentGenerationJobRunner.java

```diff
@@ -0,0 +1,267 @@
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.MapConfiguration;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.pinot.common.config.TableConfig;
+import org.apache.pinot.common.utils.DataSize;
+import org.apache.pinot.common.utils.TarGzCompressionUtils;
+import org.apache.pinot.filesystem.PinotFSFactory;
+import org.apache.pinot.ingestion.common.Constants;
+import org.apache.pinot.ingestion.common.PinotClusterSpec;
+import org.apache.pinot.ingestion.common.PinotFSSpec;
+import org.apache.pinot.ingestion.common.SegmentGenerationJobSpec;
+import org.apache.pinot.ingestion.common.SegmentGenerationTaskRunner;
+import org.apache.pinot.ingestion.common.SegmentGenerationTaskSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.filesystem.PinotFS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class SegmentGenerationJobRunner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentGenerationJobRunner.class);
+  private static final String OFFLINE = "OFFLINE";
+
+  private SegmentGenerationJobSpec _spec;
+
+  public SegmentGenerationJobRunner(SegmentGenerationJobSpec spec) {
+    _spec = spec;
+    if (_spec.getInputDirURI() == null) {
+      throw new RuntimeException("Missing property 'inputDirURI' in 'jobSpec' file");
+    }
+    if (_spec.getOutputDirURI() == null) {
+      throw new RuntimeException("Missing property 'outputDirURI' in 'jobSpec' file");
+    }
+    if (_spec.getRecordReaderSpec() == null) {
+      throw new RuntimeException("Missing property 'recordReaderSpec' in 'jobSpec' file");
+    }
+    if (_spec.getTableSpec() == null) {
+      throw new RuntimeException("Missing property 'tableSpec' in 'jobSpec' file");
+    }
+    if (_spec.getTableSpec().getTableName() == null) {
+      throw new RuntimeException("Missing property 'tableName' in 'tableSpec'");
+    }
+    if (_spec.getTableSpec().getSchemaURI() == null) {
+      if (_spec.getPinotClusterSpecs() == null || _spec.getPinotClusterSpecs().length == 0) {
+        throw new RuntimeException("Missing property 'schemaURI' in 'tableSpec'");
+      }
+      PinotClusterSpec pinotClusterSpec = _spec.getPinotClusterSpecs()[0];
+      String schemaURI = String
+          .format("%s/tables/%s/schema", pinotClusterSpec.getControllerURI(), _spec.getTableSpec().getTableName());
+      _spec.getTableSpec().setSchemaURI(schemaURI);
+    }
+    if (_spec.getTableSpec().getTableConfigURI() == null) {
+      if (_spec.getPinotClusterSpecs() == null || _spec.getPinotClusterSpecs().length == 0) {
+        throw new RuntimeException("Missing property 'tableConfigURI' in 'tableSpec'");
+      }
+      PinotClusterSpec pinotClusterSpec = _spec.getPinotClusterSpecs()[0];
+      String tableConfigURI =
+          String.format("%s/tables/%s", pinotClusterSpec.getControllerURI(), _spec.getTableSpec().getTableName());
```

Comment:
Hard-coded 'tables'? Let us at least get this from some method like generateTableConfigURI(), so that if we have versioned URIs, we know where to add/change these.